### PR TITLE
Add `BloomForQuestionAnswering`

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR adds the class `BloomForQuestionAnswering`, insipred from [`GPTJForQuestionAnswering`](https://github.com/huggingface/transformers/blob/4c962d5e790d06c142af35aad165c74c0bcf861a/src/transformers/models/gptj/modeling_gptj.py#L1024), as the community asked for the release of this class (See the discussion here: https://huggingface.co/bigscience/bloom/discussions/46#633b35f21fd49ee0b64e29d2) 

cc @sgugger @ydshieh @LysandreJik  @ArthurZucker

### CODE REVIEW

File: src/transformers/models/bloom/modeling_bloom.py [hunk 1]
```diff
@@ -1167,3 +1168,96 @@ def forward(
...
+    SQuAD (a linear layers on top of the hidden-states output to compute `span start logits` and `span end logits`).
+    """,
+    BLOOM_START_DOCSTRING,
+)
+class BloomForQuestionAnswering(BloomPreTrainedModel):
+    _keys_to_ignore_on_load_missing = [r"h.*.self_attention.scale_mask_softmax.causal_mask", r"lm_head.weight"]
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
```
Comment (younesbelkada) [inline src/transformers/models/bloom/modeling_bloom.py:1185]: It's a stupid question but it seems that for Question Answering the number of labels is always 2 (`start_logits`, `end_logits`), shall we hardcode it to 2? Maybe I am missing something here 🤔

**Review by sgugger — COMMENTED** [review]
Thanks for working on this! Should be good to merge once the labels are properly set :-)

File: src/transformers/models/bloom/modeling_bloom.py (hunk 1 above)
Comment (younesbelkada) [reply src/transformers/models/bloom/modeling_bloom.py:1185]: Perfect thanks, now it makes sense yes ! Then we should also modify it for GPTJ maybe? Happy to open a PR later for that

**Comment by younesbelkada** [comment]
I have just fixed your suggestions 💪 Gently pinging you here, and let me know if you need me to open a PR for GPTJ too ;) !

**Review by sgugger — APPROVED** [review]
Good to merge, thanks!

**Comment by younesbelkada** [comment]
Thanks a lot!!
